File: algo/extra-long-factorials.py
# Extra Long Factorials
# Calculate a very large factorial that doesn't fit in the conventional numeric data types.
#
# https://www.hackerrank.com/challenges/extra-long-factorials/problem
#

import functools
import operator

def extraLongFactorials(n):
    # Complete this function
    fac = functools.reduce(operator.mul, [i for i in range(1, n + 1)], 1)
    print(fac)
    # ou tout simplement math.factorial(n) !!!

if __name__ == "__main__":
    n = int(input().strip())
    extraLongFactorials(n)

File: algorithms/implementation/extra-long-factorials.hs
-- Extra Long Factorials
-- Calculate a very large factorial that doesn't fit in the conventional numeric data types.
--
-- https://www.hackerrank.com/challenges/extra-long-factorials/problem
--
import Control.Applicative
import Control.Monad
import System.IO

fac :: Integer -> Integer
fac 1 = 1
fac n = n * fac (n - 1)

main :: IO ()
main = do
    n_temp <- getLine
    let n = read n_temp :: Integer
    let result = fac n
    print result
